Assets/Scripts/Menu Scripts/SensorManager.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using ISAACS;
+
+public class SensorManager : MonoBehaviour {
+
+    Button thisButton;
+    Toggle thisToggle;
+    Text thisSensorText;
+    List<ROSSensorConnectionInterface> sensorList = new List<ROSSensorConnectionInterface>();
+    List<string> subscriberList;
+    ROSSensorConnectionInterface selectedSensor;
+
+    public bool leftArrow = false;
+    public bool rightArrow = false;
+    public bool displaySensorText = false;
+    //any other buttons can be added here
+
+    void Awake()
+    {
+        //Adds listener to the button, if component is button
+        if (GetComponent<Button>() != null) {
+            thisButton = GetComponent<Button>();
+            thisButton.onClick.AddListener(() => { OnClickEvent(); });
+        }
+
+        //Adds listener to toggle, if component is toggle
+        else if (GetComponent<Toggle>() != null)
+        {
+           thisToggle = GetComponent<Toggle>();
+           thisToggle.onValueChanged.AddListener(delegate { ToggleValueChanged(thisToggle); });
+        }
+
+        else
+        {
+            thisSensorText = GetComponent<Text>();
+        }
+    }
+
+
+    void OnClickEvent()
+    {
+        if (leftArrow)
+        {
+            showPreviousSensor();
+        }
+
+        if (rightArrow)
+        {
+            showNextSensor();
+        }
+
+        //any other buttons can be added here, if needed
+    }
+
+
+    // function 1: new drone selected: update all the sensors
+    // UpdateUI function: paramaters: List of Sensors
+    // store the list of sensors as current sensors
+    // selected sensor is the first sensor
+    // update the ui with the first sensor's subscribers.
+
+    public void initializeSensorUI(List<ROSSensorConnectionInterface> allSensors)
+    {
+        sensorList.AddRange(allSensors);
```

Comment:
I might be missing something obvious. 
But my understanding is that AddRange() appends onto the existing list. So if we switch drones the new list will contain the senors of the old drone + new drone?